### PR TITLE
Make planners aware of goal tolerances.

### PR DESCRIPTION
**Pull request by jasonimercer**
Created a common parent class for base_local_planner and base_global_planner.
Created a goal tolerance object.
Added goal tolerance shared pointer to new common parent class.
Populate goal tolerance object in move_base.

@skaynama  @p6chen

### CODE REVIEW

File: nav_core/include/nav_core/base_planner.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+#include "nav_core/goal_tolerances.h"
+
+namespace nav_core
+{
+/**
+ * @class BasePlanner
+ * @brief Provides an interface for planners used in navigation.
+ */
+class BasePlanner
```
Comment (skaynama) [inline nav_core/include/nav_core/base_planner.h]: Probably need a virtual destructor?

**Comment by jasonimercer** [comment]
@servos This is the base PR to add goal tolerances to the navigation stack.

**Comment by servos** [comment]
LGTM

**Comment by skaynama** [comment]
LGTM

**Pull request merged**